asv_bench/benchmarks/benchmarks.py

```diff
@@ -11,159 +11,174 @@
-        (10, 1_000_000, 10, 1_000_000),
-        (1_000_000, 10, 1_000_000, 10),
+        # the case extremely inefficient
+        # (20, 500_000, 10, 1_000_000),
+        (500_000, 20, 1_000_000, 10),
     ]
-    GROUPBY_DATA_SIZE = [
+    UNARY_OP_DATA_SIZE = [
         (5000, 5000),
         (10, 1_000_000),
         (1_000_000, 10),
     ]
-    DATA_SIZE = [(50_000, 128)]
 else:
-    MERGE_DATA_SIZE = [
-        (2000, 100, 2000, 100),
+    BINARY_OP_DATA_SIZE = [
+        (256, 256, 256, 256),
+        (20, 10_000, 10, 25_000),
+        (10_000, 20, 25_000, 10),
     ]
-    GROUPBY_DATA_SIZE = [
-        (2000, 100),
+    UNARY_OP_DATA_SIZE = [
+        (256, 256),
+        (10, 10_000),
+        (10_000, 10),
     ]
-    DATA_SIZE = [(10_000, 128)]
 
-JOIN_DATA_SIZE = MERGE_DATA_SIZE
-ARITHMETIC_DATA_SIZE = GROUPBY_DATA_SIZE
 
-CONCAT_DATA_SIZE = [(10_128, 100, 10_000, 128)]
+class TimeMultiColumnGroupby:
+    param_names = ["data_size", "count_columns"]
+    params = [
+        UNARY_OP_DATA_SIZE,
+        [6]
+    ]
 
+    def setup(self, data_size, count_columns):
+        self.df = generate_dataframe(
+            ASV_USE_IMPL, "int", data_size[1], data_size[0], RAND_LOW, RAND_HIGH
+        )
+        self.groupby_columns = [col for col in self.df.columns[:count_columns]]
 
-class TimeGroupBy:
-    param_names = ["impl", "data_type", "data_size"]
+    def time_groupby_agg_quan(self, data_size, count_columns):
+        self.df.groupby(by=self.groupby_columns).agg("quantile")
+
+    def time_groupby_agg_mean(self, data_size, count_columns):
+        self.df.groupby(by=self.groupby_columns).apply(lambda df: df.mean())
+
+
+class TimeGroupByDefaultAggregations:
+    param_names = ["data_size"]
     params = [
-        ["modin", "pandas"],
-        ["int"],
-        GROUPBY_DATA_SIZE,
+        UNARY_OP_DATA_SIZE,
     ]
 
-    def setup(self, impl, data_type, data_size):
+    def setup(self, data_size):
         self.df = generate_dataframe(
-            impl, data_type, data_size[0], data_size[1], RAND_LOW, RAND_HIGH
+            ASV_USE_IMPL, "int", data_size[1], data_size[0], RAND_LOW, RAND_HIGH
         )
+        self.groupby_column = self.df.columns[0]
 
-    def time_groupby_sum(self, impl, data_type, data_size):
-        self.df.groupby(by=self.df.columns[0]).sum()
+    def time_groupby_count(self, data_size):
+        self.df.groupby(by=self.groupby_column).count()
 
-    def time_groupby_mean(self, impl, data_type, data_size):
-        self.df.groupby(by=self.df.columns[0]).mean()
+    def time_groupby_size(self, data_size):
+        self.df.groupby(by=self.groupby_column).size()
 
-    def time_groupby_count(self, impl, data_type, data_size):
-        self.df.groupby(by=self.df.columns[0]).count()
+    def time_groupby_sum(self, data_size):
+        self.df.groupby(by=self.groupby_column).sum()
+
+    def time_groupby_mean(self, data_size):
+        self.df.groupby(by=self.groupby_column).mean()
```

Comment:
I think we should trigger materialization for groupby as well

Reply:
Will do.